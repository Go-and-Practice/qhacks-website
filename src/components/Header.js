import React, { Component } from "react";
import Menu from "./Menu";
import MobileMenu from "./MobileMenu";

let prevScrollY = 0;

class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isHeaderVisible: true,
      isMobileMenuVisible: false,
      isLogoVisible: false,
      isAtTheTop: true
    };
  }

  componentDidMount() {
    window.onscroll = () => {
      const isAtTheTop = window.scrollY === 0;
      const isLogoVisible = window.scrollY > 200;
      const isHeaderVisible =
        window.scrollY < prevScrollY || window.scrollY < 100;
      if (
        this.state.isHeaderVisible !== isHeaderVisible ||
        this.state.isLogoVisible !== isLogoVisible ||
        this.state.isAtTheTop !== isAtTheTop
      ) {
        this.setState({ isHeaderVisible, isLogoVisible, isAtTheTop });
      }
      prevScrollY = window.scrollY;
    };
  }

  onMobileMenuClicked = () => {
    this.setState({ isMobileMenuVisible: !this.state.isMobileMenuVisible });
  };

  render() {
    return (
      <div>
        <div
          css={{
            width: "100vw",
            position: "fixed",
            top: this.state.isHeaderVisible ? "0px" : "-100px",
            transition: "0.5s",
            backgroundColor: "#ffffff",
            opacity: this.state.isMobileMenuVisible ? "1.0" : "0.8",
            zIndex: "3",
            "@media(min-width:860px)": {
              backgroundColor: this.state.isAtTheTop
                ? "rgba(0,0,0,0)"
                : "#ffffff"
            }
          }}
        >
          <div
            css={{
              "@media(max-width:860px)": {
                display: "grid",
                gridTemplateColumns: "auto auto"
              }
            }}
          >
            <Menu
              menuItems={this.props.menuItems}
              imgCss={this.state.isLogoVisible}
            />
            <MobileMenu
              menuItems={this.props.menuItems}
              isMenuVisible={this.state.isMobileMenuVisible}
              toggleMenu={this.onMobileMenuClicked}
            />
          </div>
        </div>
      </div>
    );
  }
}

export default Header;
