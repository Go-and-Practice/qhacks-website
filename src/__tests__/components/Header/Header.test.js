import React from "react";
import Header from "../../../components/Header";
import { shallow } from "enzyme";

const breadCrumbs = ["b1", "b2", "b3"];
describe("Header", () => {
  it("renders correctly", () => {
    expect(shallow(<Header breadCrumbs={breadCrumbs} />)).toMatchSnapshot();
  });
});
