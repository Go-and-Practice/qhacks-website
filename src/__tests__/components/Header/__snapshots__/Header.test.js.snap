// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <div
    css={
      Object {
        "@media(min-width:860px)": Object {
          "backgroundColor": "rgba(0,0,0,0)",
        },
        "backgroundColor": "#ffffff",
        "opacity": "0.8",
        "position": "fixed",
        "top": "0px",
        "transition": "0.5s",
        "width": "100vw",
        "zIndex": "3",
      }
    }
  >
    <_default>
      <div
        css={
          Object {
            "@media(max-width:860px)": Object {
              "display": "grid",
              "gridTemplateColumns": "auto auto",
            },
          }
        }
      >
        <Menu
          imgCss={false}
          menuItems={
            Array [
              "b1",
              "b2",
              "b3",
            ]
          }
        />
        <MobileMenu
          isMenuVisible={false}
          menuItems={
            Array [
              "b1",
              "b2",
              "b3",
            ]
          }
          toggleMenu={[Function]}
        />
      </div>
    </_default>
  </div>
</div>
`;
